.github/workflows/e2e-macos.yaml

```diff
@@ -53,11 +53,11 @@ jobs:
           echo "has_creds=$has_creds" >> $GITHUB_OUTPUT
       - name: configure aws credentials
         uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
-        if: ${{ steps.vars.outputs.has_creds == true }}
+        if: ${{ steps.vars.outputs.has_creds == 'true' }}
         with:
           role-to-assume: ${{ secrets.ROLE }}
           role-session-name: credhelper-test
-          aws-region: ${{ secrets.REGION }}
+          aws-region: ${{ secrets.REGION }}        
```

Comment:
nit: here and a few other places appears to be some trailing whitespace being appended to aws-region line.

Reply:
make sense. I add new commands in makefile and only let push event and nightly runs to generate the test report.

Reply:
For merges to mainline, I summarize two conditions. If there are more conditions, please let me know.
```
if [ "${{ github.event_name }}" == "push" ] && [ "${{ github.ref }}" == "refs/heads/main" ] || 
   [ "${{ github.event_name }}" == "pull_request" ] && [ "${{ github.base_ref }}" == "main" ]; then
```